fragment.go

```diff
@@ -0,0 +1,122 @@
+package vite
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+)
+
+// HTMLFragment generates an HTML fragment for Vite integration based on the provided configuration.
+//
+// This function takes a Config struct and uses it to create the necessary HTML
+// elements for including Vite-managed assets in a web page. The returned HTML
+// is typically intended to be placed in the <head> section of an HTML document.
+//
+// Parameters:
+//   - config: A Config struct containing the necessary configuration options for Vite integration.
+//
+// Returns:
+//   - template.HTML: An HTML fragment as a template.HTML type, which can be safely embedded in HTML templates.
+//   - error: An error if the HTML fragment generation fails for any reason.
+//
+// The returned HTML fragment may include elements such as:
+//   - Script tags for Vite client in development mode
+//   - Links to stylesheets
+//   - Module preload tags
+//   - Script tags for entry points and other necessary JavaScript
+//
+// Usage example:
+//
+//	fragment, err := HTMLFragment(myConfig)
+//	if err != nil {
+//	    // Handle error
+//	}
+//	// Use fragment in your HTML template
+func HTMLFragment(config Config) (template.HTML, error) {
+	pd := &pageData{
+		IsDev:     config.IsDev,
+		ViteEntry: config.ViteEntry,
+		ViteURL:   config.ViteURL,
+	}
+
+	if config.IsDev {
+		pd.PluginReactPreamble = template.HTML(PluginReactPreamble(config.ViteURL))
```

Comment:
I agree. We could add something to `vite.Config` like this:

```go
type Config struct {

        // ...
        
	// RequiresReactPreamble specifies whether an additional preamble script
	// should be included when using React plugins. This is necessary for
	// proper integration with React-based plugins as described in the Vite
	// documentation. The default value is true, ensuring that the preamble
	// is included unless explicitly disabled.
	//
	// For more information, refer to the Vite guide on backend integration:
	// [Backend Integration - 2]: https://vitejs.dev/guide/backend-integration.html#backend-integration
	RequiresReactPreamble bool
	
	// ...
}
```

Since React might be the most commonly used Vite template, we could default to `true` if it's not specified in the config.

```go
func HTMLFragment(config Config) (template.HTML, error) {
     //...
     
     	if config.IsDev {
		// Check if the React preamble script should be included based on the
		// RequiresReactPreamble configuration. If RequiresReactPreamble is true
		// or if it is not explicitly set (i.e., its value is nil), the preamble
		// script will be included. By default, RequiresReactPreamble is true
		// due to the widespread use of React plugins.
		if config.RequiresReactPreamble == true || &config.RequiresReactPreamble == nil {
			pd.PluginReactPreamble = template.HTML(PluginReactPreamble(config.ViteURL))
		}
	} else {
             // ...
        }
        
     //...
}
```

Then, if someone is using a Vite template other than React, they can opt out of the preamble:

```go
viteFragment, err := vite.HTMLFragment(vite.Config{
      FS:                    viteFS,
      IsDev:                 config.IS_DEV,
      ViteEntry:             homeEntry,
      ViteURL:               config.VITE_URL,
      RequiresReactPreamble: false,
})
```

Reply:
Does Vite create different preambles per framework? The [Vite docs](https://vitejs.dev/guide/backend-integration.html) don't say so. So in Dev mode, can't we offer the Preamble as the Vite fragment, whereas in Production mode we return the necessary script and link tags?

Reply:
My bad, there seems to be official support in the form of an [official plugin for every major web framework](https://vitejs.dev/plugins/).

From the developer experience (DX) perspective, we should probably have this library inject the correct preamble, right? Of course we can start with React only, since that's probably the 90%+ case. But we could offer an option to automatically inject the correct preamble for every officially supported plugin. Maybe something like:

```go
frag, err := vite.HTMLFragment(vite.Config{
      FS:        viteFS,
      IsDev:     config.IS_DEV,
      ViteEntry: homeEntry,
      ViteURL:   config.VITE_URL,
      Framework: vite.React, // or, later: vite.Vue, vite.VueJSX, vite.Vue2, vite.Vue2JSX, or vite.ReactSWC
})
```

Reply:
That's fair. I only see reference to React's preamble in the docs:

> Note if you are using React with @vitejs/plugin-react, you'll also need to add this before the above scripts, since the plugin is not able to modify the HTML you are serving ...

Instead of a boolean in vite.Config, I'll implement your approach, which gives us room to grow.

Reply:
Of course, choose a better name for `Framework` if you feel like it. I'm not a native speaker.

Reply:
@olivere, no worries. 'Framework' is fine. Or perhaps 'ViteTemplate', e.g. `npm create vite@latest -- --template vue`

Reply:
I've added a feature for this issue in [33237c1](https://github.com/olivere/vite/pull/10/commits/33237c16c8afcf8c880862ad509d85f0e99ccd83)